### ISSUE TITLE
plugin-taskbar: display of icons during d&d could be improved

### ISSUE THREAD

**Issue by pmattern** [lxqt/lxqt#860]
The display of the icon shown when buttons are dragged & dropped in plugin-taskbar could be improved in two ways, IMO:

The icons are placed somewhat below the mouse pointer. So given the position of the pointer during d&d a large part is outside the screen most of the time.
Transparency seems to be missing, see black box around the icon.
![lxqt-panel_taskbar-dandd-icon-position](https://cloud.githubusercontent.com/assets/9860801/10716262/fe4eebaa-7b32-11e5-97c7-6c189b4bb1b2.png)

On a side note I'm not sure about the purpose of those two symbols "hand" and "arrow to upper left" getting displayed in parallel.

**Comment by palinek** [comment]
It's the matter of panel/taskbar placement. If the panel is on left/top edge of the screen, the icon position is OK.

Transparency is ok, if compositor is active.

**Comment by jleclanche** [comment]
We should set a different background if compositor is not active then.

**Comment by jleclanche** [comment]
There is a feedback problem though. I don't think I like the icon being shown as part of the dnd, it looks like you can drop the app eg on the desktop to create a shortcut or something. 

We could instead change the cursor. But the item *being* dragged should get a visible highlight of some kind.

**Comment by palinek** [comment]
> There is a feedback problem though. I don't think I like the icon being shown as part of the dnd, it looks like you can drop the app eg on the desktop to create a shortcut or something.

I don't agree here. If the application `accept` the drop it should know how to handle it. On my system (oxygen cursor theme) only lxqt-panel's taskbar accepts the DnD (cursor 'closed fist'), other application (desktop, lxqt-panel's other plugins...) don't accept the DnD (cursor 'arrow with red crossed circle')

**Comment by palinek** [comment]
> On a side note I'm not sure about the purpose of those two symbols "hand" and "arrow to upper left" getting displayed in parallel.

This is a matter of cursor theme used. We don't do any special logic with the cursor.

**Comment by palinek** [comment]
@pmattern Is the position of icon better in lxde/lxqt-panel#254?

**Comment by pmattern** [comment]
The location of the actual icon (like Firefox in screenshot above) is fixed by lxde/lxqt-panel#254. But I'd disagree that this issue can be closed due to this PR:

*transparency*
Transparency indeed gets available by using a compositor. But I'm not sure whether this is a desirable solution with regards to LXQt's aim to be lightweight. Something similar had been discussed regarding lxqt-notificationd [here](https://github.com/lxde/lxqt/issues/596#issuecomment-93565959).
Wouldn't it eventually be possible to display the icons the same way as in plugin-menu where transparency works very well without compositing?

*mouse pointer / symbols*
The pointer is of course set by the cursor theme but that fact that two symbols are displayed in parallel isn't and still seems questionable to me as stated above.
Aside from this I'm seeing that 'no sign' here as well as soon as the icon gets d&d'ed over the desktop. But not sure whether it's a suitable hint to prevent users from thinking it could be dropped to the desktop due to its tiny size.
![lxqt-panel_taskbar-no-sign](https://cloud.githubusercontent.com/assets/9860801/10728497/b67067cc-7be2-11e5-942e-688d44312382.png)
Basically I think having the icon displayed over the desktop as well doesn't hurt as d&d to the desktop - as well as from the panel's main menu to the desktop - may be available some day

**Comment by paulolieuthier** [comment]
>  Wouldn't it eventually be possible to display the icons the same way as in plugin-menu where transparency works very well without compositing?

It's very different. Basically, it's a Qt thing we don't have much control of. In the menu, Qt paints the icons to the menu items and displays them correctly. Regarding the taskbar's dnd icons and lxqt-notificationd's notification widgets, there nothing behide them to paint them on. We should work on it, of course, but for now people should use a compositor.

> The pointer is of course set by the cursor theme but that fact that two symbols are displayed in parallel isn't and still seems questionable to me as stated above. Aside from this I'm seeing that 'no sign' here as well as soon as the icon gets d&d'ed over the desktop. But not sure whether it's a suitable hint to prevent users from thinking it could be dropped to the desktop due to its tiny size.

It's the kind of thing users will learn quick. It's intuitive.

**Comment by pmattern** [comment]
How about omitting the actual mouse pointer ("black hand and arrow" in screenshots above) and displaying the other ones ("white arrow and no sign" ~) only during d&d?

**Comment by palinek** [comment]
> How about omitting the actual mouse pointer ("black hand and arrow" in screenshots above) and displaying the other ones ("white arrow and no sign" ~) only during d&d?

IMO we should stick to the default pointers provided by the "environment" and don't fiddle with them in any way. The shown icon is only a supporting information for user to know which window/button is he/she moving.

**Comment by pmattern** [comment]
Running lxde/lxqt-panel@1b25e94 on Arch Linux the transparency seems to be fixed as can be seen in the following screenshot taken using Openbox without any compositor active
![lxqt-panel_plugin-taskbar-drag-and-drop_fixed-transparency](https://cloud.githubusercontent.com/assets/9860801/14405384/34ae5d28-fe8e-11e5-83bc-42985e78e117.png)
(Haven't got time to look for distinct corresponding commit or correlation to Qt 5.*6*, will eventually report back should I have.)

**Comment by paulolieuthier** [comment]
So we can close this?

**Comment by pmattern** [comment]
When icons get dragged over the panel there's still that icon "hand + arrow". Besides provided by the cursor theme I still think it looks odd and a replacement should be considered.
More importantly that "no sign" icon into which the mouse pointer turns upon hovering outside lxqt-panel is really tiny. As such I figure it's rather hard for users to realize it's meant to indicate that icons cannot be dragged and dropped to the desktop.
IMO changing these two topics would improve the UX.

But these remnants are minor polish stuff that certainly can not be considered a blocker to a release any longer.
So I suggest we leave the issue open but remove the milestone.

**Comment by paulolieuthier** [comment]
I think it's safe to let the cursor theme handle that. Breeze Snow here has nice cursor icons. The distributions have the responsability to choose good cursor themes.

**Comment by palinek** [comment]
> I think it's safe to let the cursor theme handle that.

I completely agree.

**Comment by pmattern** [comment]
Meanwhile I stumbled upon another finding: The mouse curse turns into that "no sign" only when the icon gets dragged over a window. The sign does not appear when an icon gets dragged over the desktop itself.
Considering the no sign is meant to indicate that icons can *not* get dropped onto the desktop this is something that should be changed, IMO.

That aside I had another look at various icon themes and basically agree that it should be up to the icon themes to present a reasonable icon now.
But it strikes that the vast majority of icon themes does provide a no sign as tiny as the one displayed [above](https://github.com/lxde/lxqt/issues/860#issuecomment-207823496), even Breeze Snow here (the mouse icon is normally rather large but turns into a tiny arrow plus "no stop" upon dragging icons from plugin-taskbar).
I still don't think thinks this makes for a good UX.

**Comment by stefonarch** [comment]
Can this be closed? It looks all good with breeze, papirus, oxygen and others.

**Comment by tsujan** [comment]
> Can this be closed?

Yes. Qt draws DND icons, not we.
